### PR TITLE
[sonic-config-engine]: Update L2 preset for dualtor

### PR DESCRIPTION
**Pull request by theasianpianist**
Signed-off-by: Lawrence Lee <lawlee@microsoft.com>

#### Why I did it
* L2 switch mode support is needed for dual ToR devices

#### How I did it
* When generating L2 preset, check for dual ToR setting from CLI option
     * `-a '{"is_dualtor": true}'`
* When dual ToR is specified, add `subtype` field to `DEVICE_METADATA` table
* When dual ToR is specified, add `MUX_CABLE`, `TUNNEL`, and `PEER_SWITCH` tables

#### How to verify it
Run the sonic-config-engine tests

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

#### A picture of a cute animal (not mandatory but encouraged)
image

### CODE REVIEW

File: src/sonic-config-engine/tests/sample_output/py2/l2switch_dualtor.json [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    "DEVICE_METADATA": {
+        "localhost": {
+            "hwsku": "Mellanox-SN2700",
+            "subtype": "DualToR"
+        }
+    },
+    "PORT": {
+        "Ethernet0": {
+            "alias": "fortyGigE0/0",
+            "lanes": "29,30,31,32",
```
Comment (tahmed-dev) [inline src/sonic-config-engine/tests/sample_output/py2/l2switch_dualtor.json:11]: Where are the number of lanes coming from?

Comment (theasianpianist) [reply src/sonic-config-engine/tests/sample_output/py2/l2switch_dualtor.json:11]: It's loaded from the sample port config file used in the tests https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-config-engine/tests/t0-sample-port-config.ini

File: src/sonic-config-engine/config_samples.py [hunk 2]
```diff
@@ -39,11 +48,44 @@ def generate_empty_config(data):
...
 def generate_l2_config(data):
+    if 'is_dualtor' in data and data['is_dualtor']:
+        is_dualtor = True
+        data.pop('is_dualtor')
+    else:
+        is_dualtor = False
     data['VLAN'] = {'Vlan1000': {'vlanid': '1000'}}
     data['VLAN_MEMBER'] = {}
-    for port in natsorted(data['PORT']):
+    if is_dualtor:
```
Comment (tahmed-dev) [inline src/sonic-config-engine/config_samples.py:58]: I do not see a setting for Loopback2  interface. This will stall linkmgrd.

**Review by prsunny — DISMISSED** [review]
lgtm, wait for other reviewers.. also plan to update https://github.com/Azure/SONiC/wiki/L2-Switch-mode

File: src/sonic-config-engine/config_samples.py (hunk 2 above)
Comment (theasianpianist) [reply src/sonic-config-engine/config_samples.py:58]: Updated, thanks

**Review by tahmed-dev — APPROVED** [review]

**Review by lguohan — APPROVED** [review]

**Pull request merged**